src/plugin/featureaction/featureactionmanager.js

```diff
@@ -118,6 +126,176 @@ plugin.im.action.feature.Manager.prototype.initialize = function() {
 };
 
 
+/**
+ * @inheritDoc
+ */
+plugin.im.action.feature.Manager.prototype.processItemsProtected = function(
+    entryType,
+    items,
+    opt_unprocess,
+    opt_unprocessOnly) {
+  // run the actions normally
+  var configs = plugin.im.action.feature.Manager.base(
+      this,
+      'processItemsProtected',
+      entryType,
+      items,
+      opt_unprocess,
+      opt_unprocessOnly);
+
+  // notify
+  if (configs && configs.length) {
+    // once all processItems() are done, do a big notify of style and color changes
+    var config = /* @type {os.im.action.ImportActionCallbackConfig} */ ({
+      color: [],
+      labelUpdateShown: false,
+      notifyStyleChange: false,
+      setColor: false,
+      setFeaturesStyle: false
+    });
+
+    // merge the layer, source, colormodel, and label events into one
+    configs.forEach((cfg) => {
+      plugin.im.action.feature.Manager.mergeNotify_(config, cfg);
+    });
+
+    // optimize the colors to avoid overlaps (max N instead of N^2 events)
+    plugin.im.action.feature.Manager.mergeNotifyColor_(config);
+
+    // send events to synch with renderer and bins
+    plugin.im.action.feature.Manager.notify_(items, config);
+  }
+};
+
+
+/**
+ * Consolidate results of desired notification(s) from multiple FeatureActions
+ *
+ * @param {os.im.action.ImportActionCallbackConfig} config
+ * @param {os.im.action.ImportActionCallbackConfig} cfg
+ * @private
+ */
+plugin.im.action.feature.Manager.mergeNotify_ = function(config, cfg) {
```

Comment:
Nit: variable naming on merge functions should make it clear which object is the `source` and which is the `target` (or the `to` and `from` object).